src/contents/key_pair.rs

```diff
@@ -85,47 +125,30 @@ impl KeyPair {
     pub fn sign(&self, data: &[u8]) -> Result<Vec<u8>, Error> {
         match self.public_key.key_type {
             KeyType::Ed25519VerificationKey2018 => {
-                let ed = Ed25519Sha512::new();
-                ed.sign(data, &self.private_key)
-                    .map_err(|e| Error::UrsaCryptoError(e))
-            }
+                Ok(x25519(array_ref!(data, 0, 32).to_owned(), array_ref!(&self.private_key, 0, 32).to_owned()).to_vec())
```

Comment:
Ed25519 and X25519 are different operations, instead here we should use `ed25519_dalek::Signer::sign` (already implemented for `ed25519_dalek::Keypair`)

Reply:
@chunningham changed to use ed signature =)